skin/src/main/java/org/researchstack/skin/task/SmartSurveyTask.java

```diff
@@ -116,8 +122,13 @@ else if(type.equals("DateConstraints"))
         {
             answerFormat = new DateAnswerFormat(AnswerFormat.DateAnswerStyle.Date);
         }
+        else if(type.equals("DurationConstraints"))
+        {
+            answerFormat = new DurationAnswerFormat(constraints.step, constraints.durationUnit);
+        }
         else
         {
+            LogExt.e(SmartSurveyTask.class, "Survey question has answer type not supported:" + type);
```

Comment:
we can throw a RuntimeException, but then why is there that UnknownAnswerFormat at all?
We would need to ask the people who designed this...